gpm_api/tests/io/test_checks.py

```diff
@@ -0,0 +1,497 @@
+    with pytest.raises(TypeError):
+        checks.check_time(123)
+
+    # Check numpy multiple timestamp
+    with pytest.raises(ValueError):
+        checks.check_time(np.array(["2014-12-31", "2015-01-01"], dtype="datetime64[s]"))
+
+    # Test with numpy non datetime64 object
+    with pytest.raises(ValueError):
+        checks.check_time(np.array(["2014-12-31"]))
+
+
+def test_check_date() -> None:
+    """Check date/datetime object is returned from varying inputs"""
+
+    # Test a datetime object
+    res = checks.check_date(datetime.datetime(2014, 12, 31))
+    assert isinstance(res, datetime.date)
+    assert res == datetime.date(2014, 12, 31)
+
+    # Test a datetime timestamp with h/m/s/ms
+    res = checks.check_date(datetime.datetime(2014, 12, 31, 12, 30, 30, 300))
+    assert isinstance(res, datetime.date)
+    assert res == datetime.date(2014, 12, 31)
+
+    # Test a string is cast to date
+    res = checks.check_date("2014-12-31")
+    assert isinstance(res, datetime.date)
+
+    # Test a np datetime object is cast to date
+    res = checks.check_date(np.datetime64("2014-12-31"))
+    assert isinstance(res, datetime.date)
+
+    # Test None raises exception
+    with pytest.raises(ValueError):
+        checks.check_date(None)
+
+
+def test_check_start_end_time() -> None:
+    """Check start and end time are valid"""
+
+    # Test a string
+    res = checks.check_start_end_time(
+        "2014-12-31",
+        "2015-01-01",
+    )
+    assert isinstance(res, tuple)
+
+    # Test the reverse for exception
+    with pytest.raises(ValueError):
+        checks.check_start_end_time(
+            "2015-01-01",
+            "2014-12-31",
+        )
+
+    # Test a datetime object
+    res = checks.check_start_end_time(
+        datetime.datetime(2014, 12, 31),
+        datetime.datetime(2015, 1, 1),
+    )
+    assert isinstance(res, tuple)
+
+    # Test the reverse datetime object for exception
+    with pytest.raises(ValueError):
+        checks.check_start_end_time(
+            datetime.datetime(2015, 1, 1),
+            datetime.datetime(2014, 12, 31),
+        )
+
+    # Test a datetime timestamp with h/m/s/ms
+    res = checks.check_start_end_time(
+        datetime.datetime(2014, 12, 31, 12, 30, 30, 300),
+        datetime.datetime(2015, 1, 1, 12, 30, 30, 300),
+    )
+
+    # Test end time in the future
+    with pytest.raises(ValueError):
+        checks.check_start_end_time(
+            datetime.datetime(2014, 12, 31, 12, 30, 30, 300),
+            datetime.datetime(2125, 1, 1, 12, 30, 30, 300),
+        )
+
+    # Test start time in the future
+    with pytest.raises(ValueError):
+        checks.check_start_end_time(
+            datetime.datetime(2125, 12, 31, 12, 30, 30, 300),
+            datetime.datetime(2126, 1, 1, 12, 30, 30, 300),
+        )
+
+    # Check a time that is generated in another timezone but does not directly
+    # carry timezone information. This should fail if the check is done on utcnow()
```

Comment:
I would suggest to raise error if has timezone and is not UTC ;)

Reply:
Do you suggest editing `gpm_api.io.checks.check_time()`, which is the underlying function here?

Reply:
If the time in `check_time()` has a defined timezone that is not UTC, raise an error. 
Of the allowed inputs, only datetime.datetime can carry time zone (tzinfo) information.
You could add a check before the return at the end of `check_time()` and check that the datetime 'time' object has a `tz_info` which is UTC or which is not specified ;) 

Note: we currently do not allow input `pandas.Timestamp`, which can also carry time zone info. @evanjt

Reply:
I have made the check, which raises the error, and built the test cases that will test this error